docs/development_environment.mdx

```diff
@@ -50,12 +50,65 @@ If the Dev Container was set up correctly - it supports debugging by default, ou
 
 It is also possible to debug a remote Home Assistance instance (e.g., productive instance) by following the procedure described [here](https://www.home-assistant.io/integrations/debugpy/).
 
+## Setup Local Repository
+
+Visit the [Home Assistant Core repository](https://github.com/home-assistant/core) and click **Fork**.
+Once forked, setup your local copy of the source using the commands:
+
+```shell
+git clone https://github.com/YOUR_GIT_USERNAME/core.git
+cd core
+git remote add upstream https://github.com/home-assistant/core.git
+```
+
 ## Manual Environment
 
 _You only need these instructions if you do not want to use devcontainers._
 
 It is also possible to set up a more traditional development environment. See the section for your operating system. Make sure your Python version is 3.9 or later.
 
+The previous section on cloning the repository is required for all listed methods below.
+
+### Developing using Docker
+
+To setup a development environment within Docker, the supplied `Dockerfile.dev` can be used.
```

Comment:
I'm not sure if we should document this as such. As pointed out in the core PR, this Docker file is meant for use with the devcontainer specification, not as a container for dev.

This will force us to maintain it and keep it compatible in two way, for a rather smaller use case. Therefore, I think we should, not document it as such.

../Frenck

Reply:
Are you saying then, to develop for HA, you must use VSCode? If not a definitive no, then I think it's important that we at least show people how to do things. If it is VSCode or the highway, then yeah, there's little advantage, but then move the file away and 'hide' it so it cannot be used easily.

Reply:
> Are you saying then, to develop for HA, you must use VSCode?

I didn't say that. However, we don't have to document, and maintain to support every possible method out there either.

If you prefer something different, that is up to you. It doesn't mean we have to document, support or maintain your way.

../Frenck

Reply:
Well the dockerfile.dev is there, in the root. People who are not familiar with VSCode (or not allowed to use it due to cooperate reasons) see this file, and wish to use it. Having some pointers _somewhere_ is of course useful to others. I know I can figure this kind of stuff out easily.

Also, there's already quite a few methods described already. Can always add a note saying 'this is not officially supported' of course :) It's just some text in the end :)

Your call in the end, so I'll leave it up to you :)

Reply:
> Your call in the end, so I'll leave it up to you :)

I'm going to friendly decline it. If multiple interests shows up in the future, we can always reconsider.

Nevertheless, if you want to develop in a different manner that suits your more, please by all means use what you prefer. Others like Pycharm or using a dev container in pycharm aren't documented either; still used by many.

../Frenck